--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -402,25 +402,30 @@
 
 include(CTest)      # defines the BUILD_TESTING option (defaults ON)
 enable_testing()
+if(BUILD_TESTING)
 add_subdirectory(tests)
 add_subdirectory(unit-tests)
 add_subdirectory(d4_ce/unit-tests)
 add_subdirectory(http_dap/unit-tests)
+endif(BUILD_TESTING)
 
 # a target that runs only the unit tests
+if(BUILD_TESTING)
 add_custom_target(unit-test
         COMMAND ${CMAKE_CTEST_COMMAND} --output-on-failure --parallel --label-regex unit
-        DEPENDS unit-tests)            # or whatever your unit‐test target is
+        DEPENDS unit-tests)
+endif(BUILD_TESTING)            # or whatever your unit‐test target is
 
 # a target that runs only the integration tests, These can't run in parallel :(
+if(BUILD_TESTING)
 add_custom_target(integration-test
         COMMAND ${CMAKE_CTEST_COMMAND} --output-on-failure --label-regex integration
-        DEPENDS tests)                 # or your integration‐test deps
+        DEPENDS tests)
 
-# a “check” target to do unit → integration in order
 add_custom_target(check
         DEPENDS unit-test integration-test
 )
+endif(BUILD_TESTING)
 
 # Custom cleanup for coverage files, etc.
 set_directory_properties(PROPERTIES
